src/thor/bidirectional_astar.cc

```diff
@@ -934,16 +936,47 @@ std::vector<std::vector<PathInfo>> BidirectionalAStar::FormPath(GraphReader& gra
   // Special case code if the last edge of the forward path is
   // the destination edge - update the elapsed time
   if (edgelabels_reverse_[idx2].predecessor() == kInvalidLabel) {
-    // destination is on a different edge than origin
+    // destination is on a different edge than origin because of that we know that the forward path
+    // did not care about partial distance along the edge when it found the edge so what we need to do
+    // is look at the second to last edge and use the reverse path (who does care about partial
+    // distance on the destination edge) to recompute the elapsed time and cost
```

Comment:
I'm having a hard time parsing this comment. How about

```
// Destination is on a different edge than origin.
// Because of this, we know that the forward path did not care about partial distance
// along the edge when it found the edge.
// So what we need to do is look at the second to last edge and use the reverse path
// to recompute the elapsed time and cost.
// (No point in doing partial distance on the destination edge since it's not used)
```

Reply:
that description is not quite right to me. we do partial distance on the destination edge, its just already taken care of by the reverse path SetDestination function when it makes the edge label. ill try to reword it and see if you like it